pkg/armhelpers/deployer.go

```diff
@@ -0,0 +1,59 @@
+package armhelpers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/prometheus/common/log"
+)
+
+type AzureTemplateDeployer struct{}
+
+func (az *AzureClient) Validate(resourceGroup, name string, d resources.Deployment) (resources.DeploymentValidateResult, error) {
```

Comment:
There's no need for the struct here. The struct was when I thought I'd have to wrap more, I'll remove the struct line